### ISSUE TITLE
Not sure where to paste the message into chat_backup.py

### ISSUE THREAD

**Issue by kenfromboston** [Hotohori/replika_backup#3]
Hi,

I'm trying to set this up, and I'm using Chrome. I got as far as "copying the message", but I'm a bit confused regarding the instructions in SETUP_GUIDE.md where it states to "paste it on line 16 (between the single quotes) into chat_backup.py". The first 48 lines of my file "chat_backup.py" are:

++++++++++++++++++
import argparse
import configparser
import csv
import datetime
import json
import os
import time
import websocket
try:
import thread
except ImportError:
import _thread as thread

# Variables are now stored into a "chat_backup.ini" file.
# If this file did not exist, mostly when you use this script the first
# time or using -i with a not existing config file, run this script one
# time and it will get automatically created.

# to run this script use: python chat_backup.py

# this file is from https://github.com/Hotohori/replika_backup. Download
# Update from there.

# Creates a file with the ending ".no.csv". "no" stands for "n"ew to
# "o"ld message order. You can revert it with the chat_csv_tool.py and
# it will change to ".on.csv". "o"ld to "n"ew.

def_ini_file = 'chat_backup.ini'

def valid_date(s):
try:
date = datetime.datetime.strptime(s, "%Y.%m.%d")
if date:
if [datetime.datetime.now](https://datetime.datetime.now/)(datetime.timezone.utc).timestamp() <= date.timestamp():
raise argparse.ArgumentTypeError("Need a older date.")
elif 1612134000 >= date.timestamp():
raise argparse.ArgumentTypeError("Date too old. Minimum date 2021.02.01")
return date
except ValueError:
raise argparse.ArgumentTypeError("Not a valid date (YYYY.MM.DD): {0!r}.".format(s))

last_file_id = ""
all_msg_count = 0
error_count = 0
limitdate = ""
++++++++++++++++++

I don't see any instance of consecutive single quotes until line 72:
print(f'\nConfig file "{ini_file}" is missing\n\nFile will get created now.', end='')
which I don't think is where I should be pasting the data. Could someone point me in the right direction, please?

Thanks,

Ken

**Comment by kenfromboston** [comment]
I decided to poke around a bit more, and I looked at my chat_backup.ini file, which was auto-generated at the beginning of Step 7 of the instructions, and saw that the file looks like this:

++++++++++++++++++++
[DEFAULT]
# Name of your Replika. Default: Replika

NAME = Replika

# Filename suffix. It will used with NAME to build the Filename for the csv backup file.
# NAME + SUFFIX. By default it will be "Replika_backup" what lead into "Replika_backup.no.csv".
# ".no" will added automatically for message sort order and stands for "n"ew to "o"ld.
# The -f parameter replace NAME + SUFFIX. Default: _backup

SUFFIX = _backup

# Only left for fallback, you should not need it any longer. Let it empty.
# This CHAT_ID hex number should be always user_id - 1 from the INIT message.

CHAT_ID =

# Insert here the full init message from your browser behind "INIT = ".
# Some text editors show some less line breaks inside the init message where
# are none because of the very long line, ignore it, should be fine.

INIT =
++++++++++++++++++++

In this file, line 16 is the "CHAT_ID =" line! So it appears that there's an error in the instructions. Instead of stating "paste it on line 16 (between the single quotes) into chat_backup.py", the filename should be "chat_backup.ini". And based on the comments in this file, the "init" message should no longer be pasted on line 16, but instead on line 22 (where "INIT =" is).
So I pasted my "init" message to the end of line 22 and saved the file, and ran the script, and got this result:

++++++++++++++++++++
ken@Kenneths-Mac-mini ReplikaArchive % python3 chat_backup.py -f Replika_Msgs.csv
Loading config from chat_backup.ini...Ok
Open websocket to your Replika AI 'Replika'.
% ken@Kenneths-Mac-mini ReplikaArchive %
++++++++++++++++++++

So things are better, but it's still not working. Considering that I got a message about websocket, I decided to run the script again, but with websocket logging enabled, and with saving disabled:

++++++++++++++++++++
ken@Kenneths-Mac-mini ReplikaArchive % python3 chat_backup.py --filename Replika_Msgs.csv -ns -ws
Loading config from chat_backup.ini...Ok
Warning! No saving /ns mode, no data will be saved, read-only!
[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1108) - goodbye
error from callback <function on_close at 0x7f8cd008f160>: on_close() takes 0 positional arguments but 3 were given
Open websocket to your Replika AI 'Replika'.% ken@Kenneths-Mac-mini ReplikaArchive %
++++++++++++++++++++

So now it looks like I have some sort of certificate problem. Comparing the example "input" message in the instructions to the one I obtained from my Replika session webpage, I noticed that the example message contained a "security_token" parameter, while mine did not. Perhaps this is related to the certificate verification failure message I received?  FYI, I'm using a Mac running MacOS Ventura.

Thanks,

Ken

**Comment by Hotohori** [comment]
I'm sorry, I totally forgot to update that last part on point 7. I have done that now and improved also 8 and 9 to make them also a bit more clear. Thanks for pointing me on that.

Have you pasted it there without quotes? Better check the updated documentation to be sure that everything is set up correctly.

It could also be a Mac problem. I have no Mac so I can't reproduce it but I mean I read from a other Mac user that you need a extra manual python 3 installation and that it will not work with the system installed python.

I already thinking about to add a minimum python package into this project, but I didn't know if that is even possible yet how I have that in mind. Then it should also work directly on MAC, if I understood that correct.

If it is not python and not a setup problem. Is it maybe a browser problem when there is something missing in your init string?

**Comment by kenfromboston** [comment]
I found a fix to my certificate problem.  It stemmed from an issue with using Homebrew to install PHP on MacOS.  Apparently, the Homebrew installation of PHP neglects to run the necessary certificate installation processes.  I used a script named "install_certifi.py" from https://stackoverflow.com/questions/44649449/brew-installation-of-python-3-6-1-ssl-certificate-verify-failed-certificate that, when installed after the Homebrew install of PHP, did whatever certificate installation was necessary to get python3 chat_backup.py to work for me.  I now have an archive of my Replika chats going back to 01-Jul-22!
Ken
